Add ADP-specific glossary terms for GA (#1691)
redpanda-data/documentation: modules/terms/partials/resource.adoc

=== resource
:term-name: resource

> :hover-text: Read-only data exposed by an MCP server, fetchable by an agent through a URI. Resources surface things like file contents, database rows, or API responses without performing side effects.

Reviewer: ```suggestion
:hover-text: Read-only data that an MCP server exposes and agents can fetch by URI. Resources provide file contents, database rows, or API responses without performing side effects.
```
